web/models/session.ex

```diff
@@ -0,0 +1,26 @@
+defmodule PhoenixTwitter.Session do
+  alias PhoenixTwitter.User
+
+  def login(params, repo) do
+    user = repo.get_by(User, email: String.downcase(params["email"]))
+
+    case authenticate(user, params["password"]) do
+      true -> {:ok, user}
+      _    -> :error
+    end
+  end
+
+  defp authenticate(user, password) do
+    case user do
+      nil -> false
+      _   -> Comeonin.Bcrypt.checkpw(password, user.password_hash)
+    end
+  end
```

Comment:
Also you can define multiple function heads kesa mag-conditional if-else or case ka.

```
authenticate(user, params["password"]) # invoke it like this

def authenticate(nil, password), do: false
def authenticate(user, password), do: .....checkpw/2 
```